com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineDesiredStateAchievingActionsStep.java

```diff
@@ -47,23 +55,33 @@ protected StepPhase executeStep(ExecutionWrapper context) {
 
     private StepPhase attemptToExecuteStep(ExecutionWrapper execution) {
         CloudApplication app = StepsUtil.getApp(execution.getContext());
-        ApplicationStartupState currentState = computeCurrentState(execution, app);
+        CloudControllerClient client = execution.getControllerClient();
+        CloudApplication retrievedCloudApplication = client.getApplication(app.getName());
+
+        ApplicationStartupState currentState = computeCurrentState(retrievedCloudApplication);
         getStepLogger().debug(Messages.CURRENT_STATE, app.getName(), currentState);
         ApplicationStartupState desiredState = computeDesiredState(execution.getContext(), app);
         getStepLogger().debug(Messages.DESIRED_STATE, app.getName(), desiredState);
 
         Set<ApplicationStateAction> actionsToExecute = getActionsCalculator(execution.getContext()).determineActionsToExecute(currentState,
             desiredState);
+        addStageStateIfNeeded(client, retrievedCloudApplication, actionsToExecute);
         getStepLogger().debug(Messages.ACTIONS_TO_EXECUTE, app.getName(), actionsToExecute);
 
         StepsUtil.setAppStateActionsToExecute(execution.getContext(), actionsToExecute);
         return StepPhase.DONE;
     }
+    
+    private void addStageStateIfNeeded(CloudControllerClient client, CloudApplication cloudApplication, Set<ApplicationStateAction> actionsToExecute) {
+        ApplicationStager defaultApplicationStager = ApplicationStagerFactory.createApplicationStager(configuration.getPlatformType());
```

Comment:
This is no longer the default application stager. Rename the field to `applicationStager`.